terraform/terraform/evaluate.go

```diff
@@ -72,7 +48,7 @@ func (e *Evaluator) Scope(data lang.Data, self addrs.Referenceable) *lang.Scope
 	return &lang.Scope{
 		Data:     data,
 		SelfAddr: self,
-		PureOnly: e.Operation != walkApply && e.Operation != walkDestroy && e.Operation != walkEval,
+		PureOnly: true,
```

Comment:
The initial value is always used for the `e.Operation`. This affects the behavior of some functions, and it is arguable whether this `PureOnly: true` is appropriate for TFLint evaluation, but it follows current behavior.